bindings/matrix-sdk-ffi/src/timeline.rs

```diff
@@ -1042,7 +1042,7 @@ impl From<&matrix_sdk_ui::timeline::InReplyToDetails> for InReplyToDetails {
 pub enum RepliedToEventDetails {
     Unavailable,
     Pending,
-    Ready { message: Arc<Message>, sender: String, sender_profile: ProfileDetails },
+    Ready { content: Arc<TimelineItemContent>, sender: String, sender_profile: ProfileDetails },
```

Comment:
Could we store some sort of id or index that uniquely and permanently references the actual item, that would allow to *not* duplicate the item in the timeline? Then we get rid of a whole range of state duplication bugs (like the one fixed by the first commit of that PR, IIUC).

Reply:
It sounds like you want more fine-grained reactivity? As long as the API consumer expects `ObservableVector<Arc<TimelineItem>>` level updates for every affected timeline item, we'll fundamentally have to walk the list to find out which items need to be updated. Let's discuss this on Matrix.